csharp/src/Apache.Arrow.Flight/Client/FlightClient.cs

```diff
@@ -32,35 +32,35 @@ public FlightClient(ChannelBase grpcChannel)
             _client = new FlightService.FlightServiceClient(grpcChannel);
         }
```

Comment:
I know it's kind of annoying, but can you add these as overloads instead of changing the original definition? e.g
```
    public AsyncServerStreamingCall<FlightInfo> ListFlights(FlightCriteria criteria = null, Metadata headers = null) {
        return ListFlights(criteria, headers, null, CancellationToken.None)
    }
    
    public AsyncServerStreamingCall<FlightInfo> ListFlights(FlightCriteria criteria, Metadata headers, System.DateTime? deadline, CancellationToken cancellationToken = default) {
        // ...
    }
```

That way, binary compatibility is preserved between versions, which matters under some circumstances.